### ISSUE TITLE
Modal animation is inconsistent

### ISSUE THREAD

**Issue by parasharrajat** [Expensify/Expensify.cash#2335]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
  The following things need to be fixed:
  1. Search page Animations.
  2. New Group page Animations.
  3. New Chat Page Animations.
  4. Settings Page Animations.
  5. IOU Modal Page Animations.
  
## Expected Result:
 Animations for all the models & EMOJI Picker should be smooth.

## Actual Result:
 When we close RightSiderbarModal (All the Pages mentioned above), it quickly closes but the opening animation is very smooth. Sometimes, Modals are open suddenly without animation.

## Action Performed:
1. Click search which opens the search modal and notices the Opening animation.
2. Now close the modal, Modal closes quickly.

## Workaround:
Does not affect functionality.

## Platform:

 Web - :heavy_check_mark: 
 iOS - :heavy_check_mark: 
 Android - :heavy_check_mark: 
 Desktop App - :heavy_check_mark: 
 Mobile Web - :heavy_check_mark: 

**Version Number:** Latest
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/24370807/114252237-d10d9c00-99c1-11eb-872f-adfe42a34751.mp4

#### Other Video

https://expensify.slack.com/files/U01J5Q81BRP/F01SZK4SMGT/5e8bd204-81ec-4646-8d02-948b7b8048e3.webm

**Expensify/Expensify Issue URL:**

**Comment by mallenexpensify** [comment]
Thanks for reporting @parasharrajat and for the thorough details above!  
@Expensify/design can I get 👀 before assigning to a triager to make sure there's not _something_ I'm missing and/or to get any thoughts/ideas. I checked both repos for similar issues and couldn't find

**Comment by megankelso** [comment]
nothing really to add for me. I notice how abrupt it is and while I don't think it should linger or anything, a more smoother transition when closing would be a great improvement.

**Comment by michelle-thompson** [comment]
Yeah, my only comment is that all modal animations should look and feel exactly the same. Right now, all three actions seem different? Correct me if I'm wrong, maybe the recording distorts this.

**Comment by shawnborton** [comment]
Yeah, I share the same sentiment as Megan and Michelle. Let's try to standardize on the same exact modal animation (smoothly animates in, smoothly animates out) everywhere. It seems like from the screen recording we have a different animation for search than from Settings.

**Comment by mallenexpensify** [comment]
@parasharrajat this is an issue we'd like to fix. I've outlined the deliverables below, if you would like to work on it can you please propose a solution and I'll assign an engineer to review? The job will pay $250 (plus $150 bonus for reporting the issue and creating this GH).  A job will be created in Upwork once I hear back from you.  

- Update modal animation when clicking the 🔍 
- Update modal animation when clicking the Global Create (green plus button in bottom left) to start a `New Chat`
- Update modal animation when clicking the Global Create (green plus button in bottom left) to start a `New Group`
- Base all of the above off the current modal animation that happens when you click the user icon/avatar at the top to open Settings and the X in Settings to close it.

**Comment by parasharrajat** [comment]
@mallenexpensify Sorry for the delayed response. Yeah, I would like to work on this.

### Proposal

### Exit Animation for Modal

1. It is not smooth as we are running the translation for complete screen width instead it should run only for sidebar width.
https://github.com/Expensify/Expensify.cash/blob/e7330c4b5c722eca9fea4f75ac3051edf187e4bd/src/libs/Navigation/AppNavigator/modalCardStyleInterpolator.js#L10-L15
which should be
```javascript
const translateX = Animated.multiply(progress.interpolate({
        inputRange: [0, 1],
        outputRange: [isSmallScreen ? screen.width : variables.sideBarWidth, 0],
        extrapolate: 'clamp',
    }), inverted);
```    

### Entry animation breaks sometimes.

1. We have discussed this thing already and we want to roll out this for the whole app. What happens is when we focus on any input immediately while we run any animation(which is running in js thread), it causes the Animation to break.
To fix this:
We have to apply the focus only after Interactions are finished. thus we would be wrapping the manual focus logic into Interaction Manager, something like this.
```javascript
  componentDidMount() {
        InteractionManager.runAfterInteractions(() => {
            this.textInput.focus();
        });
    }
 ```   
This fixes the problem at the root. Other than this I have not noticed any glitches, if there are I would tackle them too.

**Comment by parasharrajat** [comment]
![Simulator Screen Shot - iPhone 11 - 2021-04-15 at 16 40 59](https://user-images.githubusercontent.com/24370807/114935692-7a60f180-9e09-11eb-8163-2dd76fdf6f4d.png)

**Comment by mallenexpensify** [comment]
@parasharrajat Re:
>Base all of the above off the current modal animation that happens when you click the user icon/avatar at the top to open Settings and the X in Settings to close it.
- Could you please help me understand this?

We would like the three update bullet point requests to be animated to function the same (or very similar?) as "the current modal animation that happens when you click the user icon/avatar at the top to open Settings and the X in Settings to close it."  Maybe the 'fix' is just reusing the code for opening/closing settings? (NOT an engineer, just guessing...)

**Comment by parasharrajat** [comment]
@mallenexpensify Alright. Will be done.

**Comment by mallenexpensify** [comment]
I need to assign an engineer to review your proposal, do you want me to do that _now_ to review your post above or do you want to update anything before I do?

**Comment by parasharrajat** [comment]
The above proposal is still valid. Please go ahead.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @thienlnam (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by parasharrajat** [comment]
@thienlnam Any feedback on the above proposal?

**Comment by marcaaron** [comment]
@thienlnam just letting you know that you should _not_ review this proposal and only triage to see if this is an `External` issue (looks like it is) then add the `External` label. Once it has an `Exported` label then someone will be assigned to review the proposal.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mitchexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MitchExpensify** [comment]
Hey @parasharrajat!m I've gone ahead and created the [Upwork job](https://www.upwork.com/jobs/~0183b7c4a42d4585fb) and hired you for it. I've set the price at $400 ($250 + $150 bonus for reporting the issue and creating this GH) which will be paid 7 days after merge 

Thanks!

**Comment by MitchExpensify** [comment]
Should `Exported` have assigned someone to review the proposal @marcaaron ?

**Comment by marcaaron** [comment]
Yeah it probably didn't because @thienlnam is still assigned.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Luke9389 (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by puneetlath** [comment]
I noticed that the emoji picker doesn't have animations. Would be good to add it.

**Comment by parasharrajat** [comment]
Sure @puneetlath I will pull the latest code in and add animation there.

**Comment by Luke9389** [comment]
Hey hey @parasharrajat. I've reviewed your proposal above and I'm giving you the green 🟢 light to start working on your PR.

**Comment by marcaaron** [comment]
Discussing with @Luke9389 now but @parasharrajat not really sure what the `InteractionManager` will resolve in this case as the issue is affecting `web` and it doesn't work properly there:

![Screen Shot 2021-04-21 at 8 19 21 AM](https://user-images.githubusercontent.com/32969087/115602152-50c91f80-a27a-11eb-8905-097d637e04c6.png)

https://reactnative.dev/docs/interactionmanager

What problem are we specifically trying to solve with the `InteractionManager`?

**Comment by Luke9389** [comment]
Yeesh, yea I totally missed that. Thanks for swooping in here Marc 🙇

**Comment by marcaaron** [comment]
Ok, but if `InteractionManager` has been reported in the docs to not work properly on web why does it improve anything? 
Is it any better than just doing a `setTimeout()`? 

Maybe there's some other root cause e.g. we are focusing before some long running JS is done?

**Comment by parasharrajat** [comment]
Yup, I will look for an alternative to that. Maybe we can use https://reactnavigation.org/docs/stack-navigator/#events Events to determine the start and end of the animation and set the focus.

**Comment by marcaaron** [comment]
@parasharrajat I'm sort of confused about why we ended up with `InteractionManager` in this linked PR? Aren't we more or less discussing above that we don't need to use it? Sorry if I missed something! Thanks!

**Comment by parasharrajat** [comment]
@marcaaron I used `transitionEnd` event on navigation. I have not added used `InteractionManager`. I just refactor the old code as it was missing the cancellation.
I will open another PR as the current was is merged already.

**Comment by parasharrajat** [comment]
@marcaaron It seems that in our use case the InteractionManager works fine as react-native-modal creates a task by calling `createInteractionHandle()` and clearing it after. So we can use the InteractionManager. We just need to update the `react-native-modal` to the latest minor version. If you say, I can give this a try otherwise Transition events are working fine for Modals. I am not sure what can be used for the Emoji Picker instead of `InteractionManager`. Also, there seems to be a regression https://github.com/Expensify/Expensify.cash/pull/2505#issuecomment-830481410 on applying the `FlatList` optimizations to the Emoji Picker which is needed otherwise animation is very rough. Should we handle Emoji Pickers sparately in different issue?

**Comment by marcaaron** [comment]
>Also, there seems to be a regression #2505 (comment) on applying the FlatList optimizations to the Emoji Picker which is needed otherwise animation is very rough. Should we handle Emoji Pickers sparately in different issue?

I think the two issues are separate and should at least have separate PRs?

I'm not really following what you're proposing for the `react-native-modal`. Can you maybe explain a bit more what you mean by that and how the inconsistency is related to that library and why we should update it?

**Comment by roryabraham** [comment]
Reopening this because we are reverting https://github.com/Expensify/Expensify.cash/pull/2680 in https://github.com/Expensify/Expensify.cash/pull/2833
